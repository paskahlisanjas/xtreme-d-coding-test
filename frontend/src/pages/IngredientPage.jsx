import React from 'react';
import StandardLayout from '../components/templates/StandardLayout';

const IngredientPage = () => {
  return (
    <StandardLayout>
      Welcome to Ingredient page.
    </StandardLayout>
  );
};

export default IngredientPage;
