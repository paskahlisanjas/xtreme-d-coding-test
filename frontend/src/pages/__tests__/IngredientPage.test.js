import React from 'react';
import { render } from '@testing-library/react';
import IngredientPage from '../IngredientPage';
import routeData from 'react-router';

const mockLocation = {
  pathname: '/',
  hash: '',
  search: '',
  state: ''
}

describe('IngredientPage', () => {
  it('should render everything properly', async () => {
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation)
    const { container } = render(<IngredientPage />);
    expect(container).toMatchSnapshot();
  })
});