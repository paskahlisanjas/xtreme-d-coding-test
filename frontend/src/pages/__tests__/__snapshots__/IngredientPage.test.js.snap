// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientPage should render everything properly 1`] = `
<div>
  <div
    class="ui mini segment"
  >
    <div
      class="ui secondary menu"
    >
      <div
        class="ui container"
      >
        <a
          class="active item"
        >
          Workbench
        </a>
        <a
          class="item"
        >
          Recipe
        </a>
        <a
          class="item"
        >
          Ingredient
        </a>
      </div>
    </div>
  </div>
  <div
    class="ui container"
  >
    <h1
      class="ui dividing header"
    >
      Available Ingredients
    </h1>
    <table
      class="ui celled unstackable table"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class=""
          >
            Name
          </th>
          <th
            class="center aligned"
          >
            Symbol
          </th>
          <th
            class="center aligned"
          >
            Illustration
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      />
    </table>
    <div
      class="ui fluid placeholder"
      style="height: 200px; border-radius: 5px;"
    />
  </div>
</div>
`;
